test/testdata/dsl/minitest.rb

```diff
@@ -21,11 +21,21 @@ def inside_method
         end
     end
 
+    def instance_helper; end
+
     before do
-        @foo = T.let(3, Integer)
+        @foo = T.let(3, T.nilable(Integer))
+        instance_helper
```

Comment:
(My assumption here is that this would fail if it did not correctly see `before`, `after`, and `it` in the same instance scope as `instance_helper`.)